### ISSUE TITLE
Mac OS compilation problem

### ISSUE THREAD

**Issue by ceppelli** [luanpotter/audioplayers#633]
I'm trying to run the included example on MacOs Catalina 10.15.6 with **flutter run -d macos** but the compilation stop with error like this

audioplayers/audioplayers/darwin/Classes/AudioplayersPlugin.m:745:48: error: 'setCategory:error:' **is unavailable: not available on macOS**
    **success = [[AVAudioSession sharedInstance] setCategory:AVAudioSessionCategoryPlayAndRecord error:&error];** 
In module 'AVFoundation' imported from audioplayers/audioplayers/darwin/Classes/AudioplayersPlugin.m:3:
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/System/Library/Frameworks/AVFoundation.framework/Frameworks/AVFAudio.framework/Headers/AVAudioSession.h:422:1: note: 'setCategory:error:' has been explicitly marked unavailable here
- (BOOL)setCategory:(AVAudioSessionCategory)category error:(NSError **)outError API_AVAILABLE(ios(3.0), watchos(2.0), tvos(9.0)) **API_UNAVAILABLE(macos)**;

I suppose that the plugin use the same code as IOS and it should be compiled with the support of **Apple catalyst compatibility layer**, but from what I see here the flutter build process won't enable it and the source code will be compiled as normal macos code and therefore it fails.

Did I something wrong? Should I run a different command for compiling the included example on macos?

Regards
Luca

**Comment by apgapg** [comment]
I am having following error when running on macos

```
Launching lib/main.dart on macOS in debug mode...
Building macOS application...
** BUILD FAILED **


The following build commands failed:
	CompileC /Users/ayushpgupta/StudioProjects/flutter_record_lesson/build/macos/Build/Intermediates.noindex/Pods.build/Debug/audioplayers.build/Objects-normal/x86_64/AudioplayersPlugin.o /Users/ayushpgupta/.pub-cache/hosted/pub.dartlang.org/audioplayers-0.16.1/darwin/Classes/AudioplayersPlugin.m normal x86_64 objective-c com.apple.compilers.llvm.clang.1_0.compiler
(1 failure)
Exception: Build process failed

```

**Comment by asteroid-studios** [comment]
I am having the same issue, stopping me from using MacOS

**Comment by hamedcom** [comment]
i have same issue too.

**Comment by eboye** [comment]
Same here.

**Comment by asteroid-studios** [comment]
I ended up solving my problem by switching to https://pub.dev/packages/just_audio for anyone still looking

**Comment by eboye** [comment]
@asteroid-studios great, thanks ... it works. I was only playing some notification audio so it was very easy to switch 👍

**Comment by luanpotter** [comment]
Fixed this as a side-effect of https://github.com/luanpotter/audioplayers/pull/672

**Comment by luanpotter** [comment]
Fixed and released on 0.17.0
Working perfectly on ios and macos as far as I was able to test :)
